File: Examples/Chart/WebForm/VB.NET/Examples/ChartGallery/Point/NPointDropLinesUC.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports Nevron.Examples.Framework.WebForm
Imports Nevron.GraphicsCore
Imports Nevron.Chart
Imports Nevron.Chart.WebForm

Namespace Nevron.Examples.Chart.WebForm
	Public Partial Class NPointDropLines2DUC
		Inherits NExampleUC
		Protected InflateMarginsCheckBoxBox As CheckBox
		Protected LeftAxisLeftAxisRoundToTickCheckBoxBox As CheckBox
		Protected DifferentColorsCheckBoxBox As CheckBox
		Protected ShowDataLabelsCheckBoxCheckBox As CheckBox

		Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
			nChartControl1.BackgroundStyle.FrameStyle.Visible = False

			' set a chart title
			Dim title As NLabel = nChartControl1.Labels.AddHeader("Point Chart")
			title.TextStyle.FontStyle = New NFontStyle("Times New Roman", 14, FontStyle.Italic)
			title.TextStyle.ShadowStyle.Type = ShadowType.LinearBlur

			' setup chart
			Dim chart As NChart = nChartControl1.Charts(0)
			chart.Enable3D = Enable3DCheckBox.Checked
			chart.BoundsMode = BoundsMode.Stretch

			' add interlace stripe
			Dim yScale As NLinearScaleConfigurator = CType(chart.Axis(StandardAxis.PrimaryY).ScaleConfigurator, NLinearScaleConfigurator)
			Dim stripStyle As NScaleStripStyle = New NScaleStripStyle(New NColorFillStyle(Color.Beige), Nothing, True, 0, 0, 1, 1)
			stripStyle.Interlaced = True
			stripStyle.SetShowAtWall(ChartWallType.Back, True)
			stripStyle.SetShowAtWall(ChartWallType.Left, True)
			yScale.StripStyles.Add(stripStyle)

			' hide the depth axis
			chart.Axis(StandardAxis.Depth).Visible = False

			' setup point series
			Dim point As NPointSeries = CType(chart.Series.Add(SeriesType.Point), NPointSeries)
			point.Name = "Point Series"
			point.DataLabelStyle.Visible = False
			point.InflateMargins = True

			point.AddDataPoint(New NDataPoint(23, "A"))
			point.AddDataPoint(New NDataPoint(67, "B"))
			point.AddDataPoint(New NDataPoint(47, "C"))
			point.AddDataPoint(New NDataPoint(12, "D"))
			point.AddDataPoint(New NDataPoint(56, "E"))
			point.AddDataPoint(New NDataPoint(78, "F"))

			' apply layout
			ApplyLayoutTemplate(0, nChartControl1, chart, title, Nothing)

			point.ShowVerticalDropLines = ShowVerticalDropLinesCheckBox.Checked
			point.ShowHorizontalDropLines = ShowHorizontalDropLinesCheckBox.Checked
		End Sub
	End Class
End Namespace

File: Examples/Chart/WebForm/CSharp/Examples/ChartGallery/Point/NPointDropLinesUC.ascx.cs
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Web.UI;
using System.Web.UI.WebControls;
using Nevron.Examples.Framework.WebForm;
using Nevron.GraphicsCore;
using Nevron.Chart;
using Nevron.Chart.WebForm;

namespace Nevron.Examples.Chart.WebForm
{
	public partial class NPointDropLines2DUC : NExampleUC
	{
		protected CheckBox InflateMarginsCheckBoxBox;
		protected CheckBox LeftAxisLeftAxisRoundToTickCheckBoxBox;
		protected CheckBox DifferentColorsCheckBoxBox;
		protected CheckBox ShowDataLabelsCheckBoxCheckBox;

		protected void Page_Load(object sender, EventArgs e)
		{
            nChartControl1.BackgroundStyle.FrameStyle.Visible = false;

			// set a chart title
			NLabel title = nChartControl1.Labels.AddHeader("Point Chart");
			title.TextStyle.FontStyle = new NFontStyle("Times New Roman", 14, FontStyle.Italic);
			title.TextStyle.ShadowStyle.Type = ShadowType.LinearBlur;

			// setup chart
			NChart chart = nChartControl1.Charts[0];
			chart.Enable3D = Enable3DCheckBox.Checked;
			chart.BoundsMode = BoundsMode.Stretch;

			// add interlace stripe
			NLinearScaleConfigurator yScale = (NLinearScaleConfigurator)chart.Axis(StandardAxis.PrimaryY).ScaleConfigurator;
            NScaleStripStyle stripStyle = new NScaleStripStyle(new NColorFillStyle(Color.Beige), null, true, 0, 0, 1, 1);
            stripStyle.Interlaced = true;
            stripStyle.SetShowAtWall(ChartWallType.Back, true);
            stripStyle.SetShowAtWall(ChartWallType.Left, true);
			yScale.StripStyles.Add(stripStyle);

			// hide the depth axis
			chart.Axis(StandardAxis.Depth).Visible = false;

			// setup point series
			NPointSeries point = (NPointSeries)chart.Series.Add(SeriesType.Point);
			point.Name = "Point Series";
			point.DataLabelStyle.Visible = false;
			point.InflateMargins = true;

			point.AddDataPoint(new NDataPoint(23, "A"));
			point.AddDataPoint(new NDataPoint(67, "B"));
			point.AddDataPoint(new NDataPoint(47, "C"));
			point.AddDataPoint(new NDataPoint(12, "D"));
			point.AddDataPoint(new NDataPoint(56, "E"));
			point.AddDataPoint(new NDataPoint(78, "F"));

			// apply layout
			ApplyLayoutTemplate(0, nChartControl1, chart, title, null);

			point.ShowVerticalDropLines = ShowVerticalDropLinesCheckBox.Checked;
			point.ShowHorizontalDropLines = ShowHorizontalDropLinesCheckBox.Checked;
		}
	}
}